test/e2e/knative_serving_test.go

```diff
@@ -301,3 +303,45 @@ func waitForRouteServingText(t *testing.T, caCtx *test.Context, routeDomain, exp
 		t.Fatalf("The Route at domain %s didn't serve the expected text \"%s\": %v", routeDomain, expectedText, err)
 	}
 }
+
+func testKnativeServingForGlobalProxy(t *testing.T, caCtx *test.Context) {
+	if err := test.UpdateGlobalProxy(caCtx, "http://1.2.4.5:8999"); err != nil {
+		t.Fatal("Failed to update proxy", err)
+	}
+
+	t.Run("wait for it to be ready after update", func(t *testing.T) {
+		test.GetPod(caCtx, knativeServing)
+	})
+
+	t.Run("deploy knative service after proxy update", func(t *testing.T) {
+		if _, err := test.WithServiceReady(caCtx, proxyHelloworldService, testNamespace, proxyImage); err == nil {
+			t.Fatal("Knative Service not ready", err)
+		}
+	})
+
+	cfg, err := test.GetConfiguration(caCtx, proxyHelloworldService, testNamespace)
```

Comment:
The error check is necessary here. Otherwise, `cfg.Status.Conditions` will get null pointer exception.